src/Enable.Extensions.Queuing.RabbitMQ/Internal/ConnectionManagement/RabbitMQManagedConnection.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using RabbitMQ.Client;
+
+namespace Enable.Extensions.Queuing.RabbitMQ.Internal
+{
+    internal class RabbitMQManagedConnection
+    {
+        public RabbitMQManagedConnection(IConnection connection)
+        {
+            Connection = connection;
+            Id = Guid.NewGuid();
+        }
+
+        public Guid Id { get; }
+
+        public IConnection Connection { get; private set; }
+
+        public int ReferenceCount { get; private set; } = 1;
+
+        public bool ChannelAllocationReached { get; set; }
```

Comment:
I don't believe that this property is required. We just need to catch allocation exceptions, and create new connections. I don't see the need to capture this property. Can we do away with it?

Reply:
This property ensures that connections that are fully allocated are not provided to clients.  The connection must be kept around in order for it to be disposed when no longer required.